Modularize, update, and enhance user focus of the "Architecture" part of Planning_for_Project

asteflova opened the pull request:
This PR proposes changes to modularize the first part of the Planning_for_Project guide. I'm not introducing almost any new content so what you see is (mostly) what was already in the guide before. 

In this PR, I split the original large files into small modules, edited them to fit the [mod docs guidelines](https://redhat-documentation.github.io/modular-docs/), and made minor to mid-sized changes to bring the content up-to-date. [A number of more complex requests](https://github.com/theforeman/foreman-documentation/issues?q=is%3Aissue+is%3Aopen+label%3A%22Planning+docs+audit%22) to update the content came up during reviews, but we agreed they don't need to block this PR (among other things, the reasoning is that the content that seems problematic is already present in master right now anyway).

Changelog:
* The new ToC is based on the outline from product pages [https://www.redhat.com/en/resources/introducing-red-hat-satellite-6]. This makes the Overview guide the next step in the “content journey” after product pages.
* Doc structure now focuses on product benefits (content/patch/etc. management) rather than features ("architecture", organizations, locations). This goes in line with user story-based documentation where features are just means to an end.
* Duplication has been reduced significantly. Module reuse has been introduced where possible to avoid content duplication.
* Headings have been improved to make the contents of each module clearer.
* "Architecture" is now used only for actual "system architecture".
* In some cases, older features have been deemphasized, references to more recent features are now included (Puppet/Ansible, Entitlement-based subscription/SCA). But this effort is nowhere near finished, to be continued in subsequent PRs.

Guides other than Planning_for_Project are affected too in a few places, but only to the extent required by module reuse: I made edits to the modules included in these other guides to make them fit in the new structure of the Planning_for_Project guide.

* [x] I am familiar with the [contributing](https://github.com/theforeman/foreman-documentation/blob/master/CONTRIBUTING.md) guidelines.

Please cherry-pick my commits into:

* As agreed today, let's cherry-pick down to Foreman 3.9 (Satellite 6.15)

jafiala commented:
I'd include also a rewrite of the intro. I know this is not really about modularization, but enhancing user focus is closely related IMO. Original:

> Red Hat Satellite is a system management solution that enables you to deploy, configure, and maintain your systems across physical, virtual, and cloud environments. Satellite provides provisioning, remote management, and monitoring of multiple Red Hat Enterprise Linux deployments with a single, centralized tool.

Proposed (can be further adjusted, this is just a minimal adjustment to remove self-referential and feature-centric phrasing):

`With the Red Hat Satellite system management solution, you can deploy, configure, and maintain your systems across physical, virtual, and cloud environments. Satellite is a single, centralized tool for provisioning, remote management, and monitoring of multiple Red Hat Enterprise Linux deployments.`

jafiala commented on guides/common/modules/con_content-flow-in-projectname.adoc:
```diff
@@ -0,0 +1,26 @@
+[id="Content-Flow-in-{ProjectNameID}_{context}"]
+= Content Flow in {ProjectName}
+
+Content in {ProjectName} flows from _external content sources_ to _{ProjectServer}_.
+The {ProjectServer} is the central component where you manage the content lifecycle.
+_{SmartProxyServers}_ then mirror the content to _managed hosts_.
```
This section and the following list could use more consistent capitalization.
I would put "external content sources" and "managed hosts" lowercase.

jafiala commented on guides/common/modules/con_content-views-in-projectname.adoc:
```diff
@@ -0,0 +1,46 @@
+A _content view_ defines a deliberately curated subset of content that your hosts can access.
+Each content view creates a set of repositories across each environment.
+The {ProjectServer} stores and manages these repositories.
+You can use content views to define which software versions a particular environment or {SmartProxyServer} uses.
```
Here I would suggest using a passive to improve readability. This must be a first for me.

```suggestion

```

Or maybe better

`You can define the software versions used by a particular environment or {SmartProxyServer} by using content views.`

jafiala commented on guides/common/modules/con_content-views-in-projectname.adoc:
```diff
@@ -0,0 +1,46 @@
+Each content view creates a set of repositories across each environment.
+The {ProjectServer} stores and manages these repositories.
+You can use content views to define which software versions a particular environment or {SmartProxyServer} uses.
+For example:
+
+* You can create a content view with older package versions for a Production environment and another content view with newer package versions for a Development environment.
+* You can create a content view with a package repository required by an operating system and another content view with a package repository required by an application.
+
+You can also combine content views to create a _composite content view_ for ease of management.
```
This could be improved. Intro to a list should be a full sentence. This will also help us not repeat information in each bullet. I would also bulletify the last line:

```suggestion
For example, you can create content views in the following ways:

* A content view with older package versions for a production environment and another content view with newer package versions for a Development environment.
* A content view with a package repository required by an operating system and another content view with a package repository required by an application.
* A _composite content view_ that combines content views for easier management.
```

jafiala commented on guides/common/modules/con_content-views-in-projectname.adoc:
```diff
@@ -0,0 +1,46 @@
+
+.Default Organization View
+A _Default Organization View_ is an application-controlled content view for all content that is synchronized to {Project}.
+Default Organization View useful if you want to register a host to {Project} and access content using a subscription, without manipulating content views and lifecycle environments.
```
missing verb, "by using", improve readability

```suggestion
Default Organization View is useful if you want to register a host to {Project} and access content by using a subscription and without manipulating content views and lifecycle environments.
```

jafiala commented on guides/common/modules/con_content-views-in-projectname.adoc:
```diff
@@ -0,0 +1,46 @@
+.Promoting a content view across environments
+When you promote a content view from one environment to the next environment in the application life cycle, the respective repository on {ProjectServer} updates and publishes the packages.
+
+In the following example, the repositories for Testing and Production contain the `_example_software_-1.0-0.{client-pkg-arch}.{client-pkg-ext}` package:
```
As per the RH SSG, user-replaced values should be italicized and in angled brackets:

```suggestion
In the following example, the repositories for Testing and Production contain the `_<example_software>_-1.0-0.{client-pkg-arch}.{client-pkg-ext}` package:
```

jafiala replied:
Apply below as well.

jafiala commented on guides/common/modules/con_hammer-cli-overview.adoc:
```diff
@@ -0,0 +1,26 @@
+[id="Hammer-CLI-Overview_{context}"]
+= Hammer CLI Overview
+
+Hammer is a command-line tool to configure and manage a {ProjectServer} through CLI commands or automation in shell scripts.
+Hammer also provides an interactive shell. 
+
+Benefits of using Hammer:
+
+* Shell scripts created for Hammer allow for basic task automation.
+* Output from Hammer commands can be redirected to other tools.
+* The `--debug` option can be used to test responses to API calls before applying them in a script, for example, `hammer --debug organization list`.
+* Changes in the API are automatically reflected in Hammer, while scripts using the API directly have to be updated manually.
```
This is very feature-centric and should be rewritten for user focus. Let me know if you need help with the phrasing.

asteflova replied:
@jafiala  Can you take another look at the assembly about administration tools? I rewrote it with your tips on enhancing user focus in mind.

This is the commit: https://github.com/theforeman/foreman-documentation/pull/2607/commits/3eac1ccbc65699033fe3f35aa4db4a3ae1bcee9c
Here is a link: https://theforeman-foreman-documentation-preview-pr-2607.surge.sh/nightly/Planning_for_Project/index-satellite.html#Tools-for-Administration-of-Satellite_planning

The part I'm most unsure of is the introduction to the list of benefits ("Use the <tool> for the following benefits"). Would you choose a different wording?

jafiala commented on guides/common/modules/con_hammer-cli-overview.adoc:
```diff
@@ -0,0 +1,26 @@
+
+To issue Hammer commands, a user must have access to the system on which {ProjectServer} is running.
+
+NOTE: The {ProjectWebUI} has development priority.
+Some features that are available through {ProjectWebUI} might not be available for Hammer.
```
This reads strangely for me. I would try to find some other wording, with hopefully a more positive statement.

asteflova replied:
Addressed in https://github.com/theforeman/foreman-documentation/pull/2607/commits/b640ede2acfc629e9807f0f716af97783ab1270c

jafiala replied:
Much better!

jafiala commented on guides/common/modules/con_kickstart-workflow.adoc:
```diff
@@ -0,0 +1,17 @@
+
+You can use Kickstart to automate the installation process of a {ProjectName} or {SmartProxyServer} by creating a Kickstart file that contains all the information that is required for the installation.
+
+When you run a {ProjectName} Kickstart script, the following workflow occurs:
```
How about this? It would make it clearer what "it" is.

```suggestion
When you run a {ProjectName} Kickstart script, the script performs the following actions:
```

jafiala replied:
With this wording, we might even omit the "It" from each bullet, which would make it more scannable.

jafiala commented:
Same would go for the "Red Hat Satellite provides features...", which occurs three times in the preview, "Read the following sections", which occurs two times, "enables you" 7x etc. These are clear markers of feature-centric language and we can quite easily replace them with user-focused wording.

jafiala commented on guides/common/modules/con_supported-usage-and-versions-of-project-components.adoc:
```diff
@@ -0,0 +1,5 @@
+= Supported Usage and Versions of {Project} Components
+
+[role="_abstract"]
+Sections below provide details about which scenarios and versions are supported within {Project}.
```
This is self-referential and not user-focused. I would consider something like "Find out which use cases, architectures, and versions are supported by {Project}."

jafiala commented on guides/common/modules/con_supported-client-architectures-configuration-management.adoc:
```diff
@@ -0,0 +1,14 @@
+[id="Supported-Client-Architectures-Configuration-Management_{context}"]
+= Supported Client Architectures: Configuration Management
+
+Supported combinations of major versions of {RHEL} and hardware architectures for configuration management with {Project}.
```
We should write in complete sentences. Please reword the intros to all these "Supported" modules into complete sentences. This will help users understand them better and it also will be easier to localize.

https://redhat-documentation.github.io/supplementary-style-guide/#shortdesc

asteflova replied:
Addressed with https://github.com/theforeman/foreman-documentation/pull/2607/commits/a07a25b33556ce35e1331d070a6386e12e5d7ce0

maximiliankolb commented on guides/common/modules/con_api-overview.adoc:
@asteflova @jafiala Please comment on the PR, not individual commits if possible. Commits become unreachable when rebasing or rewriting the history in other ways. When I click on the notificiation, I get this "warning" from Github: 

![image](https://github.com/theforeman/foreman-documentation/assets/12595287/1a67377e-6137-4712-8694-8c5b946a82f5)

Reviewing commit by commit however is generally a really smart way!

maximiliankolb commented on guides/common/modules/con_overview-of-hosts-in-project.adoc:
```diff
@@ -1,10 +1,8 @@
 
-[role="_abstract"]
-A host is any Linux client that {ProjectName} manages.
+A host is any Linux client that {ProjectName} manages. Hosts can be physical or virtual.
```
Please stick to "one sentence per line".

maximiliankolb commented on guides/common/modules/con_major-project-components.adoc:
```diff
@@ -0,0 +1,5 @@
+= Major {Project} Components
+
+A typical {Project} deployment consists of the following components: a {ProjectServer}, {SmartProxyServers} that mirror content from {ProjectServer}, and managed hosts that receive content from {ProjectServer} and {SmartProxyServers}.
+Learn about these components and the features they provide.
```
Very unlike the docs. Maybe reword to "the following chapters provide an overview of a typical/hypotential setup" or smiliar. Reads a bit marketing-y.

maximiliankolb commented on guides/common/modules/con_overview-of-hosts-in-project.adoc:
```diff
@@ -0,0 +1,19 @@
+[id="Overview-of-Hosts-in-{Project}_{context}"]
+= Overview of Hosts in {Project}
+
+A host is any Linux client that {ProjectName} manages.
```
Depends on the build target. orcharhino and upstream support Windows hosts AFAIK.

maximiliankolb commented on guides/common/modules/con_overview-of-hosts-in-project.adoc:
```diff
@@ -0,0 +1,19 @@
+
+With {Project}, you can manage hosts at scale, including monitoring, provisioning, remote execution, configuration management, software management, and subscription management.
+
+You can manage your hosts from the {ProjectWebUI} or from the command line.
```
also via API and FAM

asteflova replied:
Let's just delete line 19. The module is included in Planning_for_Project and Managing_Hosts and management tools are covered elsewhere in those guides.

maximiliankolb commented on guides/common/modules/con_project-server-overview.adoc:
```diff
@@ -0,0 +1,18 @@
+[id="{ProjectServerID}-Overview_{context}"]
+= {ProjectServerTitle} Overview
+
+The {ProjectServer} is the central component of a {Project} deployment where you plan and manage the content life cycle.
```
```suggestion
The {ProjectServer} is the central component of a {Project} deployment where you plan and manage the content lifecycle.
```

maximiliankolb commented on guides/common/modules/con_project-server-overview.adoc:
```diff
@@ -0,0 +1,18 @@
+
+A typical {Project} deployment includes one {ProjectServer} on which you perform the following operations:
+
+* Content life cycle management
```
```suggestion
* Content lifecycle management
```

maximiliankolb commented on guides/common/modules/con_project-server-overview.adoc:
```diff
@@ -0,0 +1,18 @@
+A typical {Project} deployment includes one {ProjectServer} on which you perform the following operations:
+
+* Content life cycle management
+* Configuration of {SmartProxyServers} and managed hosts through GUI, CLI, and API
```
I would not mix managing Smart Proxy Servers and Clients. It's also missing FAM

maximiliankolb commented on guides/common/modules/con_project-server-overview.adoc:
```diff
@@ -0,0 +1,18 @@
+
+* Content life cycle management
+* Configuration of {SmartProxyServers} and managed hosts through GUI, CLI, and API
+* System provisioning
```
Please reword "system".

maximiliankolb commented on guides/common/modules/con_project-server-overview.adoc:
```diff
@@ -0,0 +1,18 @@
+* Patch management
+* Subscription management
+
+The {ProjectServer} delegates content distribution, host provisioning, and communication to the {SmartProxyServers}.
```
We generally don't say "_the_ ProjectServer" or "_the_ SmartProxyServer".

maximiliankolb commented on guides/common/modules/con_project-server-overview.adoc:
```diff
@@ -0,0 +1,18 @@
+* Subscription management
+
+The {ProjectServer} delegates content distribution, host provisioning, and communication to the {SmartProxyServers}.
+The {ProjectServer} itself also includes a default, integrated {SmartProxyServer}.
```
```suggestion
By default, your {ProjectServer} includes an integrated {SmartProxy}.
```

maximiliankolb commented on guides/common/modules/con_project-server-overview.adoc:
```diff
@@ -0,0 +1,18 @@
+The {ProjectServer} itself also includes a default, integrated {SmartProxyServer}.
+
+{ProjectServer} also contains a fine-grained authentication system.
+You can grant {Project} operators permissions to access precisely the parts of the infrastructure for which they are responsible.
```
Maybe add a link to https://docs.theforeman.org/nightly/Administering_Project/index-katello.html#Managing_Users_and_Roles_admin

maximiliankolb commented on guides/common/modules/con_projectserver-operating-system.adoc:
```diff
@@ -0,0 +1,7 @@
+{Project} has packages for {EL} 8, Debian 11 and Ubuntu 20.04.
+Katello plug-in packages, which provide content management capabilities, are only available for {EL}.
+
+The only architecture {Project} has packages for is x86_64.
```
```suggestion
{Project} is only packaged for the x86_64 archiecture.
```

Or maybe "{Team} only packages {Project} for ..."?

Lennonka replied:
+1 for "{Team} only packages"

maximiliankolb commented on guides/common/modules/con_provisioning-management-with-projectname.adoc:
```diff
@@ -0,0 +1,7 @@
+[id="Provisioning-Management-with-{ProjectNameID}_{context}"]
+= Provisioning Management with {ProjectName}
+
+With {ProjectName}, you can provision hosts on various platforms with many provisioning methods from a centralized console.
```
I am unsure about the word "platforms" here. Maybe it's perfect for users new to Foreman? Or maybe we should stick to "compute resources" and/or "clouds"?

maximiliankolb commented on guides/common/modules/con_provisioning-management-with-projectname.adoc:
```diff
@@ -0,0 +1,7 @@
+
+With {ProjectName}, you can provision hosts on various platforms with many provisioning methods from a centralized console.
+
+Learn how {Project} simplifies host provisioning.
```
Reads a bit marketing-ish.

maximiliankolb commented on guides/common/modules/con_provisioning-methods-in-projectname.adoc:
```diff
@@ -0,0 +1,34 @@
+[id="provisioning-methods-in-{ProjectNameID}_{context}"]
```
```suggestion
[id="Provisioning-Methods-in-{ProjectNameID}_{context}"]
```

maximiliankolb commented on guides/common/modules/con_provisioning-methods-in-projectname.adoc:
```diff
@@ -0,0 +1,34 @@
+[id="provisioning-methods-in-{ProjectNameID}_{context}"]
+= Provisioning Methods in {ProjectName}
+
+Provisioning methods are part of the {ProjectName} application life cycle to create, manage, and update hosts.
```
```suggestion
Provisioning methods are part of the {ProjectName} application lifecycle to create, manage, and update hosts.
```

This does not make any sense to me. Please reword and split "Project lifecycle" and "lifecycle of managed hosts" more clearly.

asteflova replied:
I think I found a way to introduce the module without using this weird sentence. So, addressed.

maximiliankolb commented on guides/common/modules/con_remote-execution-in-projectname.adoc:
```diff
@@ -0,0 +1,42 @@
+For more information, see {ManagingHostsDocURL}Template_Writing_Reference_managing-hosts[Template Writing Reference] in _{ManagingHostsDocTitle}_.
+endif::[]
+
+Several job templates for shell scripts and Ansible are included by default.
```
Unsure if we should use "provider_type" in some way. See `rg "provider_type:"` in foreman_remote_execution.

asteflova replied:
Rerouted to https://github.com/theforeman/foreman-documentation/issues/2772

maximiliankolb commented on guides/common/modules/ref_list-of-key-open-source-components-of-projectserver.adoc:
```diff
@@ -0,0 +1,35 @@
+[role="_additional-resources"]
+.Additional resources
+* See https://access.redhat.com/articles/1343683[Satellite 6 Component Versions] for a complete list of the upstream components integrated into {Project} and for information about which upstream component versions were delivered with different versions of {Project}.
+
```
```suggestion
```

maximiliankolb commented on guides/doc-Planning_for_Project/master.adoc:
```diff
@@ -14,29 +14,36 @@ ifdef::satellite[]
+
+include::common/assembly_supported-usage-and-versions-of-project-components.adoc[leveloffset=+1]
 
 [[part-Deployment_Planning]]
 = {Project} Deployment Planning
```
Please move to a separate file in `guides/common/modules`.

asteflova replied:
The Deployment Planning part of the guide is out of scope of this PR. (But definitely a to-do, I plan to review it after I'm done with the Overview part).

maximiliankolb commented on guides/doc-Planning_for_Project/master.adoc:
OK for now, but it'd be nice to create a quick follow-up PR to move all content to `guides/common/modules/`.

asteflova replied:
This PR moves only the content from Part I of Planning_for_Project to modules/. In other words, only the parts of the guide that I'm currently getting up-to-date are modularized.

I'm leaving everything else untouched for now -- both in terms of the accuracy of that content and its location under topics/. But I intend to review it after Part I is done.

asteflova replied:
Also, rather than move it in a quick PR, I'd rather take the time to review the remaining content in topics/ properly. Considering how much outdated information is being uncovered in Part I in this PR, I expect it will be similar for the rest too.

maximiliankolb commented on guides/doc-Planning_for_Project/topics/Deployment_Considerations.adoc:
```diff
@@ -134,7 +134,13 @@ Such servers lie outside of the {Project} infrastructure, however, it is recomme
+include::../common/modules/snip_red-hat-images.adoc[]
+endif::[]
+
+image::life-cycle-satellite.png[Content Life Cycle in {ProjectName}]
```
```suggestion
image::life-cycle-satellite.png[Content Lifecycle in {ProjectName}]
```

At other places, we might also talk about the "Application Lifecycle". Reword?

asteflova replied:
I'd rather leave this untouched for now. The Deployment Planning part is out of scope of this PR.

Regarding life cycle vs lifecycle, this part of the guide uses "life cycle", so I think it's best to stay consistent for now -- until I have time to review Deployment Planning in another PR.

maximiliankolb replied:
I think this has been changed globally by Brian since my comment. See 4a14a77f1f0ba305303602fc62be1220c1b7c6e8

asteflova replied:
Ok, thank you Brian! I found 2 remaining instances of "Life cycle" so I fixed those.

But I'll leave the question of application vs content life cycle for later when I start looking at the Deployment part of the guide.

maximiliankolb commented on guides/doc-Planning_for_Project/topics/Provisioning_Concepts.adoc:
```diff
@@ -33,7 +33,7 @@ To provision machines using PXE booting, ensure that you meet the following requ
 
 * Ensure that all the network-based firewalls are configured to allow clients on the subnet to access the {SmartProxy}.
-For more information, see xref:figu-Satellite_Topology_with_Isolated_Capsule[].
+For more information, see xref:{SmartProxy}-Networking_planning[].
```
IMHO if we use "xref:" for links, then we can _always_ also use "_{context}" (if the anchor permits this).

Maybe a quick follow-up PR is the way to bring all anchors up to our usual conventions?

asteflova replied:
This concerns the Deployment Planning part, that will be reviewed in another PR.

maximiliankolb reviewed (changes requested):
reviewed the second half of the diff.

Lennonka commented on guides/common/modules/con_configuring-and-setting-up-remote-jobs.adoc:
```diff
@@ -1,7 +1,5 @@
-
-Any command that you want to apply to a remote host must be defined as a job template.
-After you have defined a job template you can execute it multiple times.
+{ProjectName} supports remote execution of commands on managed hosts.
```
I agree, however, I'm wondering if we should take this discussion into an Issue or to the dicussion board and invite developers to express their opinion on the terminology.

asteflova replied:
Issue created: https://github.com/theforeman/foreman-documentation/issues/2654

Lennonka commented on guides/common/modules/con_content-flow-in-projectname.adoc:
```diff
@@ -0,0 +1,28 @@
+
+*External content sources*::
+You can configure many content sources with {Project}.
+The supported content sources include the Red{nbsp}Hat Customer Portal, Git repositories, Ansible collections, Docker Hub, Puppet Forge, SCAP repositories, or internal data stores of your organization.
```
I'm not sure if Satellite can directly consume content from RH CP, but I'm missing a mention of RH CDN + link to the CDN structure for more info.

asteflova replied:
A module on CDN is included in Managing Content, so I added a link to that guide.

Lennonka commented on guides/common/modules/con_content-flow-in-projectname.adoc:
```diff
@@ -0,0 +1,28 @@
+You can configure many content sources with {Project}.
+The supported content sources include the Red{nbsp}Hat Customer Portal, Git repositories, Ansible collections, Docker Hub, Puppet Forge, SCAP repositories, or internal data stores of your organization.
+*{ProjectServer}*::
+On the {ProjectServer}, you plan and manage the content life cycle.
```
I don't think that "lifecycle" as one word is correct. According to the Merriam-Webster dictionary, which is recommended by the IBM SG, it should be two words.

bangelic replied:
@Lennonka Since the UI spells it out as one word, I ended up replacing it throughout documentation because it was inconsistently spelled "life cycle" and "lifecycle".  Refer to this PR: https://github.com/theforeman/foreman-documentation/pull/2633

Lennonka commented on guides/common/modules/con_api-overview.adoc:
```diff
@@ -0,0 +1,21 @@
+[id="API-Overview_{context}"]
+= API Overview
```
```suggestion
= {Project} API Overview
```

Lennonka commented:
Top-level suggestions:

1. How about splitting the two-part guide into two guides per part? For example:
    - {Project} Overview, Concepts, and Components
    - Planning {Project} Deployment

2. Missing a chapter about Configuration Management, ie. about Ansible, Puppet, and for upstream, Salt (there's a special guide about Salt CM in upstream)
    - in the old guide, we had supported client architectures for Puppet, however, the info about what architectures are supported for Ansible was entirely missing (it probably depends on Ansible?)
3. Missing information about the REX pull mode under remote execution
    - this might also need a list of client architectures that support the MQTT client

Lennonka commented on guides/common/modules/con_provisioning-methods-in-projectname.adoc:
```diff
@@ -0,0 +1,34 @@
+With {ProjectName}, you can provision hosts by using the following methods:
+
+Bare Metal Provisioning::
+{Project} provisions bare metal hosts primarily by using PXE boot and MAC address identification.
```
Well, there is the PXE-less way :)

asteflova commented on guides/doc-Planning_for_Project/images/web-ui-overview.png:
The latest version doesn't include the screenshot (I read in a comment on another PR that the convention is to avoid screenshots). Instead, I found a module in the Admin guide that describes the navigation features and tabs, so I reused that module with some tweaks.

asteflova commented on guides/common/assembly_configuring-and-setting-up-remote-jobs.adoc:
```diff
@@ -1,6 +1,6 @@
 include::modules/con_configuring-and-setting-up-remote-jobs.adoc[]
 
-include::modules/con_about-running-jobs-on-hosts.adoc[leveloffset=+1]
+include::modules/con_remote-execution-in-projectname.adoc[leveloffset=+1]
```
Thanks! I agree with using upstream terminology. Changed here and for a bunch of other file names from this PR.

asteflova commented on guides/common/assembly_supported-usage-and-versions-of-project-components.adoc:
```diff
@@ -0,0 +1,32 @@
+
+ifdef::satellite[]
+include::modules/con_supported-usage-of-project-components.adoc[leveloffset=+1]
+endif::[]
```
> Not sure if this is Satellite-only.

Some of the includes are Satellite-only, some are actually non-Satellite. I didn't change these ifdefs in this PR, and to be honest, I'd prefer not to investigate to keep this PR from growing even further.

> You also also shorten this to (for now) one big "if satellite" block.

I prefer smaller ifdef blocks rather than large ones. Because with large ones, there is a risk that someone (like me!) will add a new include in the middle without noticing the ifdefs.

So, no action taken.

asteflova commented on guides/common/modules/con_browsing-hosts-in-projectwebui.adoc:
```diff
@@ -0,0 +1,16 @@
+[id="Browsing-Hosts-in-{ProjectWebUI}_{context}"]
+= Browsing Hosts in {ProjectWebUI}
+
+In the {ProjectWebUI}, you can browse all hosts recognized by {ProjectServer}, grouped by type:
```
I think this describes the navigation tabs in the web UI under *Hosts*, in which case All, Discovered, Content, and Host Collections would appear to be correct.

asteflova commented on guides/common/modules/con_browsing-hosts-in-projectwebui.adoc:
```diff
@@ -0,0 +1,16 @@
+Alternatively, `{asterisk}node{asterisk}` will also find the content host `dev-node.example.com.`
+
+WARNING: {ProjectServer} is listed as a host itself even if it is not self-registered.
+Do not delete {ProjectServer} from the list of hosts.
```
I think this falls beyond the scope of this PR. I had to create this module in order to be split it from another one that I'm reusing in the Overview docs... (While I'm fine doing quick fixes for the content that is out of scope, this is more than a quick fix :)) So no action.

asteflova commented on guides/common/modules/con_content-flow-in-projectname.adoc:
```diff
@@ -0,0 +1,28 @@
+
+*External content sources*::
+You can configure many content sources with {Project}.
+The supported content sources include the Red{nbsp}Hat Customer Portal, Git repositories, Ansible collections, Docker Hub, Puppet Forge, SCAP repositories, or internal data stores of your organization.
```
Can you be more specific @maximiliankolb? My intention was to phrase this in a way that would fit all targets: "... content sources _include_ ...".

Looking at master, that version says that RH CP is the primary source and lists the others as alternatives: https://github.com/theforeman/foreman-documentation/blob/c50a4492d436ccd9c1634943e1a51ec714fd8723/guides/doc-Planning_for_Project/topics/Introduction.adoc?plain=1#L40  Was that better?

maximiliankolb replied:
For example, orcharhino would also support DEB content or Python content.

asteflova replied:
> But I can handle this with a follow-up PR.

I won't argue with you on that! :) Thank you, and if you want an issue to track it, here it is: https://github.com/theforeman/foreman-documentation/issues/2770

asteflova commented on guides/common/modules/con_content-flow-in-projectname.adoc:
```diff
@@ -0,0 +1,28 @@
+On the {ProjectServer}, you plan and manage the content life cycle.
+*{SmartProxyServers}*::
+By creating {SmartProxyServers}, you can establish content sources in various geographical locations.
+*Managed hosts*::
```
Moved to https://github.com/theforeman/foreman-documentation/issues/2654

asteflova commented on guides/common/modules/con_introduction-to-provisioning.adoc:
```diff
@@ -1,2 +1,5 @@
 = Introduction to Provisioning
+
+Provisioning is a process that starts with a bare physical or virtual machine and ends with a fully configured, ready-to-use operating system.
+Using {ProjectName}, you can define and automate fine-grained provisioning for a large number of hosts.
```
Moved to https://github.com/theforeman/foreman-documentation/issues/2654

mjivraja commented on guides/common/modules/con_smartproxy-overview.adoc:
```diff
@@ -1,9 +1,7 @@
-With {SmartProxyServers}, you can extend the reach and scalability of your {Project} deployment.
-
-{SmartProxyServers} provide the following functionalities in a {ProjectName} deployment:
+With {SmartProxyServers}, you can extend the reach and scalability of your {Project} deployment. {SmartProxyServers} provide the following functionalities in a {ProjectName} deployment:
```
```suggestion
With {SmartProxyServers}, you can extend the reach and scalability of your {Project} deployment.
{SmartProxyServers} provide the following functionalities in a {ProjectName} deployment:
```
Recalling that meme you posted :smile:

asteflova commented on guides/common/modules/con_supported-client-architectures-configuration-management.adoc:
```diff
@@ -0,0 +1,14 @@
+[id="Supported-Client-Architectures-Configuration-Management_{context}"]
+= Supported Client Architectures: Configuration Management
```
That's a great point, thank you. I replaced the colon with "for". (I think leveloffset would unnecessarily introduce additional complexity to ToC, I prefer to keep my ToCs as flat as possible.)

asteflova commented on guides/doc-Planning_for_Project/master.adoc:
```diff
@@ -14,29 +14,36 @@ ifdef::satellite[]
-[[part-Architecture]]
-= {Project} Architecture
+[id="Project_Overview_and_Concepts_{context}"]
+= {Project} Overview and Concepts
```
I'm also leaning towards splitting Planning_for_Project into two guides. In its current state, the guide mixes conceptual overview information and practical steps to be taken during deployment, which is.. not great. But it will have to wait after everything has been modularized. It will be easier to do the splitting once we have proper assemblies.

asteflova replied:
> Can we add an attribute for this guide? if so, we could use it here. :)

Let's not do that for now and wait until/if we split this into two guides. So no action.

mjivraja commented on guides/common/modules/con_api-overview.adoc:
```diff
@@ -0,0 +1,21 @@
+Scripts based on API commands communicate directly with the {Project} API, which makes them faster than scripts based on Hammer commands or Ansible playbooks relying on modules within {ansible-namespace}.
+
+IMPORTANT: API commands differ between versions of {Project}.
+When you prepare to upgrade {ProjectServer}, update any scripts that contain {Project} API commands.
```
```suggestion
When you prepare to upgrade {ProjectServer}, update all the scripts that contain {Project} API commands.
```
No?

mjivraja commented on guides/common/modules/con_client-operating-systems.adoc:
```diff
@@ -0,0 +1,27 @@
+[id="Client-Operating-Systems_{context}"]
+= Client Operating Systems
+
+{Project} can help you manage any kind of operating systems that have clients {Project} can integrate with.
```
This sentence sounds bit confusing.

mjivraja commented on guides/common/modules/con_client-operating-systems.adoc:
```diff
@@ -0,0 +1,27 @@
+* Katello host tools
+* Tracer utility
+
+For Red Hat family operating systems, SELinux must not be set to disabled mode.
```
```suggestion
For Red Hat family operating systems, SELinux must be set to enforcing or permissive mode.
```
Can this be better?

mjivraja commented on guides/common/modules/con_client-operating-systems.adoc:
```diff
@@ -0,0 +1,27 @@
+The Katello plug-in provides functionality for content and subscription management.
+The following utilities are provided for selected CentOS and SLES operating systems:
+
+* Subscription manager
+* Katello host tools
+* Tracer utility
```
Alphabetical order?

mjivraja commented on guides/common/modules/con_content-and-patch-management-with-foreman.adoc:
```diff
@@ -0,0 +1,12 @@
+endif::[]
+
+ifdef::katello,satellite,orcharhino[]
+Learn how {Project} simplifies content and patch management.
+For details on how to manage content with {Project}, see {ContentManagementDocURL}[_{ContentManagementDocTitle}_].
```
```suggestion
For more information on how to manage content and patch with {Project}, see {ContentManagementDocURL}[_{ContentManagementDocTitle}_].
```
Keeping a bit straight.

mjivraja commented on guides/common/modules/con_content-delivery-network-structure.adoc:
```diff
@@ -1,5 +1,9 @@
+= Content Delivery Network Structure
+
+ifdef::katello,orcharhino[]
+NOTE: The information in this section applies if you use a Red{nbsp}Hat Manifest to consume Red{nbsp}Hat content.
```
```suggestion
NOTE: Information in this section applies only if you consume Red[nbsp]Hat content using a Red{nbsp}Hat Manifest.
```
Does this sound better to you?

mjivraja reviewed (commented):
Second set of suggestions. Along with some queries for my learning.

asteflova commented on guides/common/assembly_configuring-and-setting-up-remote-jobs.adoc:
```diff
@@ -1,6 +1,6 @@
 include::modules/con_configuring-and-setting-up-remote-jobs.adoc[]
 
-include::modules/con_about-running-jobs-on-hosts.adoc[leveloffset=+1]
+include::modules/con_remote-execution-in-foreman.adoc[leveloffset=+1]
```
@maximiliankolb suggested to use upstream terminology rather than attribute keys in file names, hence the difference between file name and ID here. I'll keep this comment thread open in case you want to discuss it (I think I'm fine either way).

mjivraja replied:
@maximiliankolb This file is also part of the downstream. Using `foreman` in the filename is new to me. Can you explain in short?

maximiliankolb replied:
Because file names are AFAIK never shown/referenced in built docs that we offer to customers, I think it's a good compromise/convention to stick to upstream terminology. This clarifies things a lot if we go with "foreman-server", "smart-proxy", and "smart-proxy-server" everywhere instead of "project" and "proxy". I particularly dislike "proxy" because some people think "HTTP proxy".

This issue/comment is not blocking your PR. I'd also be happy to give it a try after you have merged this. I consider this a small QoL change for maintainers :)

asteflova commented on guides/common/modules/con_web-ui-overview.adoc:
```diff
@@ -0,0 +1,26 @@
+| *Content* | Provides content management tools.
+This includes content views, activation keys, and life cycle environments.
+| *Hosts* | Provides host inventory and provisioning configuration tools.
+| *Configure* | Provides general configuration tools and data, including host groups and Ansible content.
```
I don't think it's a good idea to document all the navigation features. IMO it doesn't serve much purpose, and it would just get outdated again (the list in this PR is not new, I'm just [reusing a module](https://docs.theforeman.org/nightly/Administering_Project/index-foreman-el.html#Navigation_Tabs_in_the_Web_UI_admin) that's already in the Admin guide).

Instead, I rephrased the introduction to clarify that this is just an overview.

Lennonka commented on guides/common/modules/con_smartproxy-networking.adoc:
```diff
@@ -0,0 +1,40 @@
+
+.{Project} topology with hosts connecting directly to {ProjectServer}
+
+In this topology, hosts connect to {ProjectServer} rather than an isolated {SmartProxy}.
```
In that case, we should probably explain in Conventions and in this guide what the difference between SmartProxy and SmartProxy Server are. You know, clarify the terms.

asteflova replied:
> We should probably explain in Conventions and in this guide what the difference between SmartProxy and SmartProxyServer is. You know, clarify the terms.

Let's have that conversation in https://github.com/theforeman/foreman-documentation/issues/2774

As for this PR, I removed references to "internal", but only in text for now. "Internal" is used in diagrams too, but there are more issues to be fixed there and I might need to reach out to designers for a proper rework of the graphics. So I'll handle that in a separate PR later.

For now, I think I can mark this thread as resolved. Reopen if you think I missed something.

asteflova commented:
Rebase after returning to this PR after ~2 months.

Left to do: Several comments, I will implement some of them, but divert others to issues or separate PRs.

I'd like to keep this PR limited to editing and only necessary updates from now on because it's been dragging on for too long at this point. The main benefit should be improved readability and a structure that supports adding descriptions of new functionality more easily than the previous structure. (Reminder: I have not added almost any new content, I'm just using what was in the guide before.)

maximiliankolb commented on guides/common/modules/con_major-project-components.adoc:
```diff
@@ -0,0 +1,4 @@
+[id="Major-{Project}-Components_{context}"]
+= Major {Project} components
+
+A typical {Project} deployment consists of the following components: a {ProjectServer}, {SmartProxyServers} that mirror content from {ProjectServer}, and managed hosts that receive content from {ProjectServer} and {SmartProxyServers}.
```
```suggestion
A typical {Project} deployment consists of the following components: a {ProjectServer}, {SmartProxyServers} that mirror content from {ProjectServer}, and hosts that receive content from {ProjectServer} and {SmartProxyServers}.
```

You only mention "receive content". Maybe also "receive their configuration"? Or what about REX?

maximiliankolb commented on guides/common/modules/con_major-project-components-additional-resources.adoc:
```diff
@@ -0,0 +1,5 @@
+[role="_additional-resources"]
+= Additional resources
```
Again: I worry that the anchor is not really unique. I lean towards ".Additional resources".

maximiliankolb commented on guides/common/modules/con_managing-project-with-ansible-collections.adoc:
```diff
@@ -2,4 +2,10 @@
 
 {Project} Ansible Collections is a set of Ansible modules that interact with the {Project} API.
-You can use {Project} Ansible Collections to manage and automate many aspects of {Project}.
+You can manage and automate many aspects of {Project} with {Project} Ansible Collections.
```
```suggestion
You can manage and automate many aspects of {Project} with {Project} Ansible collections.
```

maximiliankolb commented on guides/doc-Planning_for_Project/topics/Deployment_Considerations.adoc:
```diff
@@ -131,10 +131,16 @@ Such servers lie outside of the {Project} infrastructure, however, it is recomme
-A *lifecycle environment* represents a stage in the content lifecycle, a *Content View* is a filtered set of content, and can be considered as a defined subset of content.
-By associating Content Views with lifecycle environments, you make content available to hosts in a defined way.
-See xref:figu-Content_Lifecycle_{context}[] for visualization of the process.
+{Project} provides features for precise management of the content life cycle.
```
```suggestion
{Project} provides features for precise management of the content lifecycle.
```

maximiliankolb commented on guides/doc-Planning_for_Project/topics/Deployment_Considerations.adoc:
```diff
@@ -131,10 +131,16 @@ Such servers lie outside of the {Project} infrastructure, however, it is recomme
-By associating Content Views with lifecycle environments, you make content available to hosts in a defined way.
-See xref:figu-Content_Lifecycle_{context}[] for visualization of the process.
+{Project} provides features for precise management of the content life cycle.
+A *life cycle environment* represents a stage in the content life cycle, a *Content View* is a filtered set of content, and can be considered as a defined subset of content.
```
```suggestion
A *lifecycle environment* represents a stage in the content lifecycle, a *Content View* is a filtered set of content, and can be considered as a defined subset of content.
```

maximiliankolb commented on guides/doc-Planning_for_Project/topics/Deployment_Considerations.adoc:
```diff
@@ -131,10 +131,16 @@ Such servers lie outside of the {Project} infrastructure, however, it is recomme
-See xref:figu-Content_Lifecycle_{context}[] for visualization of the process.
+{Project} provides features for precise management of the content life cycle.
+A *life cycle environment* represents a stage in the content life cycle, a *Content View* is a filtered set of content, and can be considered as a defined subset of content.
+By associating Content Views with life cycle environments, you make content available to hosts in a defined way.
```
```suggestion
By associating Content Views with lifecycle environments, you make content available to hosts in a defined way.
```

maximiliankolb commented on guides/common/modules/con_web-ui-overview.adoc:
```diff
@@ -0,0 +1,25 @@
+= {ProjectWebUI} overview
+
+You can manage and monitor your {Project} infrastructure from a browser with the {ProjectWebUI}.
+For example, you can use the following navigation features in the {ProjectWebUI}
```
```suggestion
For example, you can use the following navigation features in the {ProjectWebUI}:
```

maximiliankolb reviewed (changes requested):
I have reviewed 37/63; will continue soon.

asteflova commented on guides/common/assembly_provisioning-management-with-project.adoc:
I tried to tie the structure to other Satellite materials at redhat.com, see "Provisioning management" and others [here](https://www.redhat.com/en/technologies/management/satellite).  So "provisioning management" makes sense in the context of RH downstream for consistency. I'd actually like to keep it this way but I'm all ears if there are concerns about accuracy etc. :)

maximiliankolb commented on guides/common/modules/con_kickstart-workflow.adoc:
Maybe also change the title to "Install {target} using Kickstart file"?

asteflova replied:
Let's track this in https://github.com/theforeman/foreman-documentation/issues/2764

maximiliankolb commented on guides/common/modules/con_organizations-and-locations-in-foreman.adoc:
```diff
@@ -0,0 +1,15 @@
+[id="Organizations-and-Locations-in-{ProjectNameID}_{context}"]
+= Organizations and locations in {ProjectName}
+
+On the {ProjectServer}, you can define multiple _organizations_ and _locations_ to help organize content, managed hosts, and configurations.
```
```suggestion
On your {ProjectServer}, you can define multiple _organizations_ and _locations_ to help organize content, hosts, and configurations.
```

AFAIK it's never "the ProjectServer"; always "your ProjectServer".

maximiliankolb commented on guides/common/modules/con_organizations-and-locations-in-foreman.adoc:
```diff
@@ -0,0 +1,15 @@
+On the {ProjectServer}, you can define multiple _organizations_ and _locations_ to help organize content, managed hosts, and configurations.
+
+Organizations::
+Organizations typically represent different business units, departments, or teams, such as Finance, Marketing, or Web Development Team.
```
Do we want/need some emphasis for the examples? I'm leaning towards italics.

asteflova replied:
IBM Style has a good section on when (not) to add emphasis... but it's not really conclusive for this particular situation. Anyway, let's try it. Applied here and elsewhere.

maximiliankolb commented on guides/common/modules/con_overview-of-hosts-in-foreman.adoc:
```diff
@@ -0,0 +1,17 @@
+A host is any Linux client that {ProjectName} manages.
+Hosts can be physical or virtual.
+
+ifndef::satellite,orcharhino[]
+You can deploy virtual hosts on any platform supported by {ProjectName}, such as Amazon EC2, Google Compute Engine, KVM, libvirt, Microsoft Azure, OpenStack, {oVirt}, Proxmox, Rackspace Cloud Services, or VMware vSphere.
+endif::[]
+ifdef::satellite[]
+You can deploy virtual hosts on any platform supported by {ProjectName}, such as Amazon EC2, Google Compute Engine, KVM, libvirt, Microsoft Azure, OpenStack, {oVirt}, Rackspace Cloud Services, or VMware vSphere.
+endif::[]
+ifdef::orcharhino[]
+You can deploy virtual hosts on any platform supported by {ProjectName}, such as Amazon EC2, Google Compute Engine, libvirt, Microsoft Azure, Oracle Linux Virtualization Manager, oVirt, Proxmox, RHV, and VMware vSphere.
+endif::[]
```
```suggestion
You can deploy virtual hosts on any platform supported by {ProjectName},
ifndef::satellite,orcharhino[]
such as Amazon EC2, Google Compute Engine, KVM, libvirt, Microsoft Azure, OpenStack, {oVirt}, Proxmox, Rackspace Cloud Services, or VMware vSphere.
endif::[]
ifdef::satellite[]
such as Amazon EC2, Google Compute Engine, KVM, libvirt, Microsoft Azure, OpenStack, {oVirt}, Rackspace Cloud Services, or VMware vSphere.
endif::[]
ifdef::orcharhino[]
such as Amazon EC2, Google Compute Engine, libvirt, Microsoft Azure, Oracle Linux Virtualization Manager, oVirt, Proxmox, RHV, and VMware vSphere.
endif::[]
```

maximiliankolb commented on guides/common/modules/con_project-server-overview.adoc:
```diff
@@ -0,0 +1,23 @@
+
+* Content lifecycle management
+* Configuration of {SmartProxyServers}
+* Configuration of managed hosts
```
```suggestion
* Configuration of hosts
```

maximiliankolb commented on guides/common/modules/con_project-server-overview.adoc:
```diff
@@ -0,0 +1,23 @@
+* Subscription management
+
+{ProjectServer} delegates content distribution, host provisioning, and communication to {SmartProxyServers}.
+{ProjectServer} itself also includes a default, integrated {SmartProxyServer}.
```
```suggestion
{ProjectServer} itself also includes a {SmartProxy}.
```

alternative: "{ProjectServer} itself also acts as s {SmartProxyServer}."

maximiliankolb commented on guides/common/modules/con_project-server-overview.adoc:
```diff
@@ -0,0 +1,23 @@
+{ProjectServer} itself also includes a default, integrated {SmartProxyServer}.
+
+{ProjectServer} also contains a fine-grained authentication system.
+You can grant {Project} operators permissions to access precisely the parts of the infrastructure for which they are responsible.
```
operators > users?

asteflova replied:
Applied.

maximiliankolb commented on guides/common/modules/con_project-server-overview.adoc:
```diff
@@ -0,0 +1,23 @@
+You can grant {Project} operators permissions to access precisely the parts of the infrastructure for which they are responsible.
+
+.Additional resources
+
```
```suggestion
```

maximiliankolb commented on guides/common/modules/con_project-server-overview.adoc:
```diff
@@ -0,0 +1,23 @@
+
+.Additional resources
+
+* See {AdministeringDocURL}Managing_Users_and_Roles_admin[_Managing Users and Roles_] in {AdministeringDocTitle} for information about managing permissions.
```
```suggestion
* For more information about managing permissions, see {AdministeringDocURL}Managing_Users_and_Roles_admin[Managing Users and Roles] in _{AdministeringDocTitle}_.
```

maximiliankolb commented on guides/common/modules/con_provisioning-management-with-foreman-additional-resources.adoc:
```diff
@@ -0,0 +1,4 @@
+[role="_additional-resources"]
+= Additional resources
```
Again, I worry about the uniqueness of this anchor. Maybe go with ".Additional resources" and a content-specific anchor?

maximiliankolb commented on guides/common/modules/con_provisioning-management-with-foreman.adoc:
```diff
@@ -0,0 +1,4 @@
+[id="Provisioning-Management-with-{ProjectNameID}_{context}"]
+= Provisioning management with {ProjectName}
+
+With {ProjectName}, you can provision hosts on various compute resources with many provisioning methods from a centralized console.
```
"centralized console" > unified interface? There's more than console as in "CLI", but also API, FAM, and Web UI.

asteflova replied:
Applied.

maximiliankolb commented on guides/common/modules/con_provisioning-methods-in-foreman.adoc:
```diff
@@ -0,0 +1,34 @@
+When provisioning bare metal hosts with {Project}, you can do the following:
++
+* Create host entries and specify the MAC address of the physical host to provision.
+* Boot blank hosts to use the {Project} discovery service, which creates a pool of hosts that are ready to be provisioned.
```
```suggestion
* Boot blank hosts to use the {Project} Discovery service, which creates a pool of hosts that are ready to be provisioned.
```

maximiliankolb commented on guides/common/modules/con_remote-execution-in-project.adoc:
```diff
@@ -0,0 +1,42 @@
+
+[NOTE]
+====
+Any {SmartProxyServer} base operating system is a client of the internal {SmartProxy} in the {ProjectServer}, and therefore this section applies to any type of host connected to {ProjectServer}, including {SmartProxies}.
```
```suggestion
Any {SmartProxyServer} base operating system is a client of the internal {SmartProxy} in your {ProjectServer}, and therefore this section applies to any type of host connected to {ProjectServer}, including {SmartProxyServers}.
```

maximiliankolb commented on guides/common/modules/con_remote-execution-workflow.adoc:
```diff
@@ -1,6 +1,9 @@
 [id="remote-execution-workflow_{context}"]
 = Remote Execution Workflow
 
+For custom Ansible roles that you create, or roles that you download, you must install the package containing the roles on your {SmartProxyServer}.
```
You can also place Ansible roles using `scp` or `wget` onto your SmartProxyServer. I am not sure if mentioning "packages" is really necessary. 

I like that you mention "you have to place them onto your SmartProxyServer" before you can/have to import them.

asteflova replied:
Let's track this in https://github.com/theforeman/foreman-documentation/issues/2766

maximiliankolb commented on guides/common/modules/con_simple-content-access-sca-in-foreman.adoc:
```diff
@@ -0,0 +1,19 @@
+ifdef::satellite[]
+[role="_additional-resources"]
+.Additional resources
+
```
```suggestion
```

maximiliankolb commented on guides/common/modules/con_smartproxy-features.adoc:
```diff
@@ -0,0 +1,76 @@
+= {SmartProxy} Features
+
+{SmartProxyServers} provide local host management services.
+With the Katello plug-in, they can also mirror content from {ProjectServer}.
```
For orcharhino and Satellite, "katello" is included by default. So I would not mention is by name. Instead, I'd say "SmartProxyServers can mirror content from ProjectServer".

For upstream, I am unsure. I thought we have moved away from notes like "if you have the Katello plug-in installed, ...". On the other hand, for an overview, I would be very much OK with mentioning Katello ...

Maybe someone can wants to weigh in? 

See also https://github.com/theforeman/foreman-documentation/pull/2607/files#r1420661842

asteflova replied:
I added an ifdef for orcharhino and satellite as suggested above, and left the original text under an ifndef.
@maximiliankolb Can you please check https://github.com/theforeman/foreman-documentation/pull/2607/commits/cbd0cede2a9b382ea24f529c510b939661cec234? Does this make sense?

maximiliankolb replied:
Yes, cbd0cede2a9b382ea24f529c510b939661cec234 LGTM!

maximiliankolb commented on guides/common/modules/con_smartproxy-networking.adoc:
```diff
@@ -0,0 +1,40 @@
+[id="{SmartProxy}-Networking_{context}"]
+= {SmartProxy} networking
+
+The communication between {ProjectServer} and managed hosts registered to a {SmartProxyServer} is routed through that {SmartProxyServer}.
```
```suggestion
The communication between {ProjectServer} and hosts registered to a {SmartProxyServer} is routed through that {SmartProxyServer}.
```

maximiliankolb commented on guides/common/modules/con_smartproxy-networking.adoc:
```diff
@@ -0,0 +1,40 @@
+= {SmartProxy} networking
+
+The communication between {ProjectServer} and managed hosts registered to a {SmartProxyServer} is routed through that {SmartProxyServer}.
+{SmartProxyServer} can also provide {Project} services to managed hosts.
```
```suggestion
{SmartProxyServer} also provides {Project} services to hosts.
```

maximiliankolb commented on guides/common/modules/con_smartproxy-networking.adoc:
```diff
@@ -0,0 +1,40 @@
+.{Project} topology with hosts connecting directly to {ProjectServer}
+
+In this topology, hosts connect to {ProjectServer} rather than an isolated {SmartProxy}.
+This situation applies also to {SmartProxies} themselves because the external {SmartProxy} is a client of {ProjectServer}.
```
Please reword "this situation".

also "client" > host

external SP > SmartProxyServer

asteflova replied:
Addressed.

maximiliankolb commented on guides/common/modules/con_smartproxy-networking.adoc:
```diff
@@ -0,0 +1,40 @@
+In this topology, hosts connect to {ProjectServer} rather than an isolated {SmartProxy}.
+This situation applies also to {SmartProxies} themselves because the external {SmartProxy} is a client of {ProjectServer}.
+
+ifndef::satellite[]
```
```suggestion
ifndef::satellite,orcharhino[]
```

maximiliankolb commented on guides/common/modules/con_smartproxy-overview.adoc:
```diff
@@ -0,0 +1,12 @@
+With {SmartProxyServers}, you can extend the reach and scalability of your {Project} deployment.
+{SmartProxyServers} provide the following functionalities in a {ProjectName} deployment:
+
+* Mirroring content from {ProjectServer} to establish content sources in various geographical locations.
```
```suggestion
* Mirroring content from {ProjectServer} to establish content sources in various geographical and logical locations.
```

maybe "logical" for multiple networks (DMZs) in the same rack/data center?

asteflova replied:
Applied.

maximiliankolb commented on guides/common/modules/con_smartproxy-overview.adoc:
```diff
@@ -0,0 +1,12 @@
+{SmartProxyServers} provide the following functionalities in a {ProjectName} deployment:
+
+* Mirroring content from {ProjectServer} to establish content sources in various geographical locations.
+By assigning a host system to a {SmartProxyServer}, you can configure this host to pull content and configuration from the {SmartProxy} in their location instead of from the central {ProjectServer}.
```
pull > receive?

asteflova replied:
Applied.

maximiliankolb commented on guides/common/modules/con_smartproxy-overview.adoc:
```diff
@@ -0,0 +1,12 @@
+By assigning a host system to a {SmartProxyServer}, you can configure this host to pull content and configuration from the {SmartProxy} in their location instead of from the central {ProjectServer}.
+* Running localized services to discover, provision, control, and configure hosts.
+
+You can specify the exact subset of content that {SmartProxyServer} makes available to hosts by using content views.
```
Maybe reword to "You can use content views to specify the exact subset of content that {SmartProxyServers} provide to hosts"?

asteflova replied:
Applied.

maximiliankolb commented on guides/common/modules/con_subscription-management-with-foreman-additional-resources.adoc:
```diff
@@ -0,0 +1,4 @@
+[role="_additional-resources"]
+= Additional resources
```
Same as always :see_no_evil:

maximiliankolb commented on guides/common/modules/con_supported-client-architectures-for-configuration-management.adoc:
```diff
@@ -0,0 +1,14 @@
+
+You can use the following combinations of major versions of {RHEL} and hardware architectures for configuration management with {Project}.
+
+.Puppet Agent Support
```
this heading does not match the file heading and introduction.

asteflova replied:
Applied.

maximiliankolb commented on guides/common/modules/con_supported-usage-and-versions-of-foreman-components-additional-resources.adoc:
```diff
@@ -0,0 +1,4 @@
+[roles="_additional-resources"]
+= Additional resources
```
same here.

maximiliankolb commented on guides/common/modules/con_supported-usage-and-versions-of-foreman-components.adoc:
feels a bit like a duplicate of guides/common/modules/con_supported-usage-of-foreman-components.adoc

asteflova replied:
I don't think so:

* con_supported-usage-and-versions-of-project-components.adoc is the introduction to the assembly (a one-liner, based on https://redhat-documentation.github.io/modular-docs/#forming-assemblies ).
* con_supported-usage-of-foreman-components.adoc is a module included in that assembly.

Normally the assembly introduction would be directly in the assembly_xyz.adoc, but in this project, assemblies are only lists of includes.

At least it makes sense to me like this. In RH downstream, this file structure gets the docs to look the way we need them to look when built. I'm happy to have a conversation, though, if there are any concerns.

maximiliankolb commented on guides/common/modules/con_supported-usage-of-foreman-components.adoc:
```diff
@@ -0,0 +1,50 @@
+
+{RHELServer}:: 
+Each {ProjectName} subscription includes one supported instance of {RHELServer}.
+Reserve this instance solely for the purpose of running  {ProjectName}.
```
```suggestion
Reserve this instance solely for the purpose of running {ProjectName}.
```

maximiliankolb commented on guides/common/modules/con_supported-usage-of-foreman-components.adoc:
```diff
@@ -0,0 +1,50 @@
+
+Foreman::
+You can extend Foreman with plug-ins packaged with {ProjectName}.
+See https://access.redhat.com/articles/1343683[Satellite 6 Component Versions] in Red{nbsp}Hat Knowledgebase for information about supported Foreman plug-ins.
```
```suggestion
ifdef::satellite[]
See https://access.redhat.com/articles/1343683[Satellite 6 Component Versions] in Red{nbsp}Hat Knowledgebase for information about supported Foreman plug-ins.
endif::[]
```

maximiliankolb commented on guides/common/modules/con_supported-usage-of-foreman-components.adoc:
```diff
@@ -0,0 +1,50 @@
+You can extend Foreman with plug-ins packaged with {ProjectName}.
+See https://access.redhat.com/articles/1343683[Satellite 6 Component Versions] in Red{nbsp}Hat Knowledgebase for information about supported Foreman plug-ins.
++
+Not supported: Extending Foreman with plug-ins in the {ProjectName} Optional repository.
```
is "{ProjectName} Optional" the name of the repo? If so, maybe add some emphasis?

asteflova replied:
Applied.

maximiliankolb commented on guides/common/modules/con_supported-usage-of-foreman-components.adoc:
```diff
@@ -0,0 +1,50 @@
+When you run the {Project} installation program, you can install and configure Puppet servers as part of {SmartProxyServers}.
+A Puppet module, running on a Puppet server on your {ProjectServer} or any {SmartProxyServer}, is also supported by {Team}.
+
+.Additional resources
+* Red{nbsp}Hat supports many different scripting and other frameworks.
+See https://access.redhat.com/articles/369183[How does Red{nbsp}Hat support scripting frameworks] in Red{nbsp}Hat Knowledgebase.
```
```suggestion
ifdef::satellite[]
.Additional resources
* Red{nbsp}Hat supports many different scripting and other frameworks.
See https://access.redhat.com/articles/369183[How does Red{nbsp}Hat support scripting frameworks] in Red{nbsp}Hat Knowledgebase.
endif::[]
```

maximiliankolb reviewed (changes requested):
finished the review based on the diff in Github. Mostly LGTM; thanks for starting this big task!

maximiliankolb commented on guides/common/modules/con_simple-content-access-sca-in-project.adoc:
```diff
@@ -0,0 +1,18 @@
+
+Simple Content Access (SCA) in {Project} simplifies management of software entitlements.
+With SCA, you can add valid subscriptions to a subscription allocation or manifest and refresh within {Project}.
+Therefore, with SCA enabled, you do not need to attach subscriptions individually to managed hosts.
```
```suggestion
Therefore, with SCA enabled, you do not need to attach subscriptions individually to hosts.
```

asteflova commented on guides/common/modules/con_supported-usage-of-project-components.adoc:
```diff
@@ -0,0 +1,48 @@
+= Supported Usage of Project Components
+
+Usage of all {ProjectName} components is supported within the context of {ProjectName} only as described below.
+Third-party usage of any components is not supported.
```
I'll just drop it, the line above is kenough.

asteflova commented on guides/common/modules/con_supported-usage-of-project-components.adoc:
```diff
@@ -0,0 +1,48 @@
+SELinux::
+Ensure SELinux is in enforcing or permissive mode.
++
+Not supported: Installation with disabled SELinux.
```
I kinda prefer my version: I chose it to ensure 100% parallelism across all entries (it's faster to scan quickly).

maximiliankolb commented on guides/common/modules/con_organizations-and-locations-in-project.adoc:
```diff
@@ -0,0 +1,15 @@
+Locations::
+Locations typically represent physical locations, such as countries or cities.
++
+By creating locations, you can define geographical sites where systems are located.
```
```suggestion
By creating locations, you can define geographical sites where hosts are located.
```

maximiliankolb commented on guides/common/modules/con_smartproxy-overview.adoc:
```diff
@@ -0,0 +1,12 @@
+{SmartProxyServers} provide the following functionalities in a {ProjectName} deployment:
+
+* Mirroring content from {ProjectServer} to establish content sources in various geographical or logical locations.
+By assigning a host system to a {SmartProxyServer}, you can configure this host to receive content and configuration from the {SmartProxy} in their location instead of from the central {ProjectServer}.
```
```suggestion
By registering a host to a {SmartProxyServer}, you can configure this host to receive content and configuration from the {SmartProxy} in their location instead of from the central {ProjectServer}.
```

maximiliankolb commented on guides/common/modules/con_major-project-components.adoc:
```diff
@@ -0,0 +1,4 @@
+[id="Major-{Project}-Components_{context}"]
+= Major {Project} components
+
+A typical {Project} deployment consists of the following components: a {ProjectServer}, {SmartProxyServers} that mirror content from {ProjectServer}, and managed hosts that receive content and configuration from {ProjectServer} and {SmartProxyServers}.
```
```suggestion
A typical {Project} deployment consists of the following components: a {ProjectServer}, {SmartProxyServers} that mirror content from {ProjectServer}, and hosts that receive content and configuration from {ProjectServer} and {SmartProxyServers}.
```

asteflova commented on guides/common/modules/con_hammer-cli-overview.adoc:
```diff
@@ -0,0 +1,25 @@
+= Hammer CLI overview
+
+You can configure and manage your {ProjectServer} with CLI commands by using Hammer.
+
```
When thinking of paragraphs, I don't think of how many sentences that para has, but how many ideas it conveys. So roughly 1 idea = 1 para.

* In this case, the first paragraph is an introduction to the module (because every [module needs an intro](https://redhat-documentation.github.io/modular-docs/#creating-concept-modules)).
* The second paragraph (line 6) marks the beginning of the actual contents of the module by introducing a bulleted list.

So while seeing a single-sentence paragraph makes me somewhat uncomfortable too, that "1 idea = 1 paragraph" guideline takes precedence. Especially here, if one tries to adhere to the mod docs templates (which I do because they are my babies).

maximiliankolb replied:
works for me!

asteflova commented on guides/common/modules/con_content-delivery-network-structure.adoc:
```diff
@@ -1,5 +1,9 @@
-[[chap-Architecture_Guide-Content_Delivery_Network_Structure]]
-== Content Delivery Network Structure
+[id="Content_Delivery_Network_Structure_{context}"]
+= Content Delivery Network Structure
```
Applied.

asteflova commented on guides/common/modules/con_projectserver-operating-system.adoc:
```diff
@@ -0,0 +1,7 @@
+[id="ProjectServer-Operating-System_{context}"]
+= {ProjectServer} operating system
```
Let's track this in https://github.com/theforeman/foreman-documentation/issues/2765

asteflova commented on guides/common/modules/con_provisioning-methods-in-projectname.adoc:
```diff
@@ -0,0 +1,34 @@
+When provisioning bare metal hosts with {Project}, you can do the following:
++
+* Create host entries and specify the MAC address of the physical host to provision.
+* Boot blank hosts to use the {Project} discovery service, which creates a pool of hosts that are ready to be provisioned.
```
I checked that this is part of the original content (Provisioning_Hosts in this case). So I think it's okay to leave it as it is for now, and I've created an issue to clarify it in a subsequent PR: https://github.com/theforeman/foreman-documentation/issues/2771

asteflova commented on guides/common/modules/con_remote-execution-in-projectname.adoc:
```diff
@@ -0,0 +1,42 @@
+[id="Remote-Execution-in-{Project}_{context}"]
+= Remote Execution in {ProjectName}
+
+With remote execution, you can run jobs on hosts remotely from {SmartProxies} using shell scripts or Ansible tasks and playbooks.
```
I think this is not that necessary in overview docs, and this modules links to more detailed guides from additional resources. So no action.

asteflova commented on guides/common/modules/con_remote-execution-in-projectname.adoc:
```diff
@@ -0,0 +1,42 @@
+
+[NOTE]
+====
+Any {SmartProxyServer} base operating system is a client of the internal {SmartProxy} in the {ProjectServer}, and therefore this section applies to any type of host connected to {ProjectServer}, including {SmartProxies}.
```
Let's try dropping it.

asteflova commented on guides/common/modules/con_supported-client-architectures-configuration-management.adoc:
```diff
@@ -0,0 +1,14 @@
+[id="Supported-Client-Architectures-Configuration-Management_{context}"]
+= Supported Client Architectures: Configuration Management
+
+You can use the following combinations of major versions of {RHEL} and hardware architectures for configuration management with {Project}.
```
Tracked in https://github.com/theforeman/foreman-documentation/issues/2773

asteflova commented on guides/common/modules/con_smartproxy-features.adoc:
```diff
@@ -0,0 +1,76 @@
+endif::[]
+ifndef::satellite[]
+If you have the Katello plug-in installed, you can use {SmartProxy} to mirror content from {ProjectServer}:
+endif::[]
```
I tried to address this, conversation happening in https://github.com/theforeman/foreman-documentation/pull/2607#discussion_r1420660300

asteflova commented on guides/common/modules/con_supported-client-architectures-content-management.adoc:
```diff
@@ -0,0 +1,16 @@
+[id="Supported-Client-Architectures-Content-Management_{context}"]
+= Supported Client Architectures: Content Management
+
+You can use the following combinations of major versions of {RHEL} and hardware architectures for registering and managing hosts with {Project}.
```
Headings changed, extending for upstream tracked in https://github.com/theforeman/foreman-documentation/issues/2773

asteflova commented on guides/common/modules/con_supported-client-architectures-host-provisioning.adoc:
Headings changed, extending for upstream tracked in https://github.com/theforeman/foreman-documentation/issues/2773

asteflova commented on guides/common/modules/con_supported-usage-of-project-components.adoc:
```diff
@@ -0,0 +1,48 @@
+Each {ProjectName} subscription includes one supported instance of {RHELServer}.
+Reserve this instance solely for the purpose of running  {ProjectName}.
++
+Not supported: Using the operating system included with {Project} to run other daemons, applications, or services within your environment.
```
> Katello on RHEL: needs subscription for RHEL system
> 
> orcharhino on AlmaLinux 8: needs orcharhino subscription
> 
> Katello on CentOS Stream 8: does not need any subscription.

I created https://github.com/theforeman/foreman-documentation/issues/2775 to review the upstream and downstream modules.

maximiliankolb commented on guides/common/modules/con_remote-execution-in-project.adoc:
```diff
@@ -0,0 +1,37 @@
+For more information, see {ManagingHostsDocURL}Template_Writing_Reference_managing-hosts[Template Writing Reference] in _{ManagingHostsDocTitle}_.
+endif::[]
+
+Several job templates for shell scripts and Ansible are included by default.
```
```suggestion
By default, {Project} includes several job templates for shell scripts and Ansible.
```

maximiliankolb commented on guides/common/modules/con_smartproxy-features.adoc:
```diff
@@ -0,0 +1,78 @@
+endif::[]
+
+Repository synchronization::
+{SmartProxyServers} pull content for selected lifecycle environments on the {ProjectServer} and make this content available to the hosts they manage.
```
```suggestion
{SmartProxyServers} pull content for selected lifecycle environments from {ProjectServer} and make this content available to the hosts they manage.
```

IMHO we try to avoid saying "the {Project...}".

maximiliankolb commented on guides/common/modules/con_smartproxy-features.adoc:
```diff
@@ -0,0 +1,78 @@
+{SmartProxyServers} pull content for selected lifecycle environments on the {ProjectServer} and make this content available to the hosts they manage.
+
+Content delivery::
+Hosts configured to use {SmartProxyServer} download content from that {SmartProxy} rather than from the central {ProjectServer}.
```
```suggestion
Hosts configured to use {SmartProxyServer} download content from that {SmartProxy} rather than from {ProjectServer}.
```

maximiliankolb reviewed (approved):
some very minor suggestions. rest LGTM.

asteflova commented:
>     1. How about splitting the two-part guide into two guides per part? For example:
>        
>        * {Project} Overview, Concepts, and Components
>        * Planning {Project} Deployment
>        
>        We can do this in another PR of course.

The first step would be to update and modularize the second part of Planning_for_Project (this PR targets only the first part of the guide). Then let's go wild and split it.

> 
>     2. Missing a chapter about Configuration Management, ie. about Ansible, Puppet, and for upstream, Salt (there's a special guide about Salt CM in upstream)
>        
>        * in the old guide, we had supported client architectures for Puppet agent, however, the info about what architectures are supported for Ansible was entirely missing (it probably depends on Ansible, but we should include it for completeness)

Let's track this in https://github.com/theforeman/foreman-documentation/issues/2776.
 
>     3. Missing information about the REX pull mode under remote execution
>        
>        * this might also need a list of client architectures that support the MQTT client

Let's track this in https://github.com/theforeman/foreman-documentation/issues/2777.

asteflova commented:
@maximiliankolb Can you take a look at https://github.com/theforeman/foreman-documentation/pull/2607/commits/d50eed40e4f7cb53fe3ffd76cc83df7559bc1041? Based on today's mod docs mtg, I added comments with information on which modules are (re)used where. What do you think -- yes, no, or...?

asteflova commented:
I updated the PR description based on what we talked about with @maximiliankolb. TL;DR: Rather than get the guide to a 100% up-to-date state in this PR (which would take forever), we're fine merging this to master (because this PR doesn't really change the content, this PR makes edits for readability and structure) and keep making further updates in subsequent (smaller) PRs.

maximiliankolb commented:
> @maximiliankolb Can you take a look at [d50eed4](https://github.com/theforeman/foreman-documentation/commit/d50eed40e4f7cb53fe3ffd76cc83df7559bc1041)? Based on today's mod docs mtg, I added comments with information on which modules are (re)used where. What do you think -- yes, no, or...?

I think we should not do this/rely on this because we use GHA and local builds as "source of truth". If you make a change, you have to consider the whole Foreman docs anyway. IMHO it's more work to maintain an extra list of "where do we use this file?" and I fear that it'll be out of sync quickly.

But I would accept this change if you/the team sees value in it.

Also, I didn't dare to make yet another comment; but one file in your diff contains "managed host" instead of just "host". 

@asteflova Big :+1: to getting this into master better sooner than later. We can always iterate.

maximiliankolb reviewed (approved):
Thanks Anet, LGTM. I am happy to merge this as is. More feedback is welcome; note that we can always iterate :)

asteflova commented:
> > @maximiliankolb Can you take a look at [d50eed4](https://github.com/theforeman/foreman-documentation/commit/d50eed40e4f7cb53fe3ffd76cc83df7559bc1041)? Based on today's mod docs mtg, I added comments with information on which modules are (re)used where. What do you think -- yes, no, or...?
> 
> I think we should not do this/rely on this because we use GHA and local builds as "source of truth". If you make a change, you have to consider the whole Foreman docs anyway. IMHO it's more work to maintain an extra list of "where do we use this file?" and I fear that it'll be out of sync quickly.

Sure, removed, it makes sense. I mostly just wanted to see what it felt like (and it _was_ a pain to add manually...).

> 
> But I would accept this change if you/the team sees value in it.
> 
> Also, I didn't dare to make yet another comment; but one file in your diff contains "managed host" instead of just "host".

:D:D There you go: https://github.com/theforeman/foreman-documentation/pull/2607/commits/3a9670f558564199443a7da537bbf92127a5d812

> 
> @asteflova Big 👍 to getting this into master better sooner than later. We can always iterate.

maximiliankolb commented:
@asteflova Please rebase to master and resolve the merge conflict.

As discussed, I will merge and cherry-pick to 3.9 afterwards :rocket:

asteflova commented:
@maximiliankolb I fixed up capitalization in 2 headings, and rebased + resolved a conflict. Feel free to merge and cherry-pick as discussed.

maximiliankolb commented:
@asteflova I think there's a typo in a file name of an image.
Can you please check "grep" and "find" `lifecycle-satellite.png`? See https://github.com/theforeman/foreman-documentation/actions/runs/7873997102/job/21482579093?pr=2607

maximiliankolb reviewed (approved):
Thanks Anet, LGTM!

maximiliankolb commented:
Merged to master and cherry-picked to 3.9: :heavy_check_mark: 
   99b97b73a2..aacbb91cb8  3.9 -> 3.9
